### PR TITLE
v8 compressed textures dds support

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
Support for DDS textures in v8

some tiny tweaks to KTX

### CODE REVIEW

**Review by Zyie — APPROVED** [review]

File: src/compressed-textures/dds/parseDDS.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const offPfFlags = 20;
+const offPfFourCC = 21;
+const offRGBBitCount = 22;
+const offRBitMask = 23;
+const offGBitMask = 24;
+const offBBitMask = 25;
+
+export function parseDDS(buffer: ArrayBuffer, supportedFormats: TEXTURE_FORMATS[]): TextureSourceOptions<Uint8Array[]>
+{
+    const header = new Int32Array(buffer, 0, headerLengthInt);
```
Comment (achamas-playco) [inline src/compressed-textures/dds/parseDDS.ts:59]: Just wondering. If this function throws, does the caller have a strategy? How would a user safeguard loading faulty textures at runtime? Can they catch this with `Asset.load`?

**Review by achamas-playco — APPROVED** [review]

File: src/compressed-textures/ktx/loadKTX.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const loadKTX = {
         priority: LoaderParserPriority.High,
     },
 
-    name: 'loadKTX2',
+    name: 'loadKTX',
```
Comment (achamas-playco) [inline src/compressed-textures/ktx/loadKTX.ts:22]: Other code still seems to be referencing `ktx2*`. Is the idea to just use the latest version for this identifier?

**Review by achamas-playco — APPROVED** [review]

File: src/compressed-textures/dds/parseDDS.ts (hunk 1 above)
Comment (GoodBoyDigital) [reply src/compressed-textures/dds/parseDDS.ts:59]: yoo! yes, this error should get caught by the Loader

File: src/compressed-textures/ktx/loadKTX.ts (hunk 2 above)
Comment (GoodBoyDigital) [reply src/compressed-textures/ktx/loadKTX.ts:22]: will do a pass to make sure every thing in code named KTX rather than KTX2 as both are supported

**Pull request merged**